data/markdown/pages/learn/accelerate/xm-cloud/pre-development/information-architecture/publishing-to-edge.md

```diff
@@ -0,0 +1,195 @@
+
+### Custom Contents Resolvers
+
+Contents Resolvers are used with the Sitecore Layout Service to provide more complex data beyond the serialization of a component data source. A Contents Resolver gets executed when the page (layout) is rendered (in the GraphQL response it is the “rendered” property). This occurs in different contexts: in Experience Editor, in the Experience Preview. In these instances, the context is known because XM Cloud parses the request.
+
+This is not possible in Experience Edge - Edge is able to execute “custom” code, so no Contents Resolvers can run on it. So it gets executed at publishing time when the runtime context (eg. from visitor’s browser interactions) is not known.
+
+Due to this, Custom Content Resolvers are not supported in XM Cloud - XM Cloud resolvers such as the Navigation Contents Resolver, that should be utilized but custom Content Resolvers should not be created. If you are migrating from XM/XP, review these resolves and plan where best to transfer.
+
+### Integrated GraphQL
+
+Integrated GraphQL is used to modify the default layout data format returned for a specific component, it is applied to renderings and it is used by the Layout Service. This gives more control over the data the front-end receives.
+
+An Integrated GraphQL query could use additional data sources that may reside in any location of the site, the publishing pipeline is not aware of what are the related items, it executes the query and builds the layout, this operation is done at publishing time and occurs on XM Cloud and the data items are picked up from the master database, not from the Experience Edge.
+
+This process is different in the [new Connector architecture](https://developers.sitecore.com/changelog/xm-cloud/18042024/faster-publishing-in-xm-cloud). The new architecture resolves the query at request time on the Edge (then it is cached). For this reason, related items need to be there to have the expected result. For example, if the query uses a non-children data source like the following:
+
+```
+query MyComponentQuery($datasource: String!, $contextItem: String!, $language: String!) {
+  ...omissis...
+  extraFields: item(path: "/sitecore/content/<coll>/<site>/Home/About/Data/Text 1", language: "en") {
+    name
+    mFields: fields {
+      name
+      jsonValue
+    }
+  }
+}
+```
+
+It is required to already have published - “/sitecore/content/<coll>/<site>/Home/About/Data/Text 1” . The connector is not aware about the relationships, so it cannot publish integrated GraphQL related items at publishing time.
```

Comment:
These HTML tags need to be escaped as they are breaking the build

Reply:
fixed